backend/src/services/imap/ImapEmailsFetcher.ts

```diff
@@ -346,4 +497,14 @@ export default class ImapEmailsFetcher {
     await this.imapConnectionProvider.cleanPool(); // Do it async because it may take up to 30s to close
     return this.isCompleted;
   }
+
+  /**
+   * Checks if the body of the email is in HTML format.
+   * @param body - The email body to check.
+   * @returns true if the body contains HTML, false if it doesn't.
+   */
+  private static isHTMLBody(body: string): boolean {
```

Comment:
Follow ` Single Responsibility Principle (SRP)`. The class `ImapEmailsFetcher`  should only contain the logic to fetch email from imap like it's name suggest, it should not contain any other logic about signature or parsing. Extract this method into a helper function and use it